eth/stagedsync/stage_log_index.go

```diff
@@ -477,25 +493,26 @@ func pruneLogIndex(logPrefix string, tx kv.RwTx, tmpDir string, pruneFrom, prune
 				return fmt.Errorf("receipt unmarshal failed: %w, block=%d", err, binary.BigEndian.Uint64(k))
 			}
 
-			notToPrune := false // To identify whether this log key has addr in noPruneContracts
+			toPrune := true
 			for _, l := range logs {
-				// If any of the logs have an address in noPruneContracts, then the whole tx is related to it, and must not be pruned, including addr and topic indexes
-				if noPruneContracts != nil && noPruneContracts[l.Address] || notToPrune {
-					notToPrune = true
-					continue
+				// No logs (or sublogs) for this txId should be pruned
+				// if one of the logs belongs to noPruneContracts lis
+				if noPruneContracts != nil && noPruneContracts[l.Address] {
+					toPrune = false
+					break
 				}
-				for _, topic := range l.Topics {
-					if err := topics.Collect(topic.Bytes(), nil); err != nil {
+			}
+			if toPrune {
+				for _, l := range logs {
+					for _, topic := range l.Topics {
+						if err := topics.Collect(topic.Bytes(), nil); err != nil {
+							return err
+						}
+					}
+					if err := addrs.Collect(l.Address.Bytes(), nil); err != nil {
 						return err
 					}
 				}
-				if err := addrs.Collect(l.Address.Bytes(), nil); err != nil {
-					return err
-				}
-			}
-			// No logs (or sublogs) for this txId should be pruned
-			// if one of the logs belongs to noPruneContracts list
-			if !notToPrune {
 				pruneLogKeyCollector.Collect(k, nil)
```

Comment:
while we are here we should maybe add a err check for this missed err here from `pruneLogKeyCollector.Collect(k, nil)`

Reply:
Added err check